core/src/main/java/com/google/instrumentation/trace/HttpPropagationUtil.java

```diff
@@ -0,0 +1,149 @@
+ *     <li>trace-id: 16-byte array representing the trace_id.
+ *     <li>span-id: 8-byte array representing the span_id.
+ *     <li>trace-options: 4-byte array representing the trace_options.
+ *     <li>Valid value example:
+ *         <ul>
+ *         <li>"0040414243444546474848494a4b4c4d4e4f616263646566676800000001"
+ *         <li>version_id = 0;
+ *         <li>trace_id = {64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79}
+ *         <li>span_id = {97, 98, 99, 100, 101, 102, 103, 104};
+ *         <li>trace_options = {0, 0, 0, 1};
+ *         </ul>
+ *     </ul>
+ *
+ * </ul>
+ *
+ * <p>All characters in the header value must be lower case and US-ASCII encoded.
+ *
+ * <p>It is strongly encouraged to use this format when using HTTP as a RPC transport.
+ *
+ * <p>Example of usage on the client:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracer.getTracer();
+ * void onSendRequest() {
+ *   try (NonThrowingCloseable ss = tracer.spanBuilder("Sent.MyRequest")) {
+ *     String headerName = HttpPropagationUtil.getHttpHeaderName();
+ *     String headerValue = HttpPropagationUtil.toHttpHeaderValue(span.context());
+ *     headers.add(headerName, headerValue);
+ *     // Send the HTTP request and wait for the response.
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example of usage on the server:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracer.getTracer();
+ * void onRequestReceived() {
+ *   String headerName = HttpPropagationUtil.getHttpHeaderName();
+ *   SpanContext spanContext = HttpPropagationUtil.fromHttpHeaderValue(headers.find(headerName));
+ *   try (NonThrowingCloseable ss =
+ *            tracer.spanBuilderWithRemoteParent(spanContext, "Recv.MyRequest").startScopedSpan() {
+ *     // Handle request and send response back.
+ *   }
+ * }
+ * }</pre>
+ */
+public final class HttpPropagationUtil {
+  // The name of the HTTP header.
+  private static final String HTTP_HEADER_NAME = "Trace-Context";
+  private static final byte VERSION_ID = 0;
+  // The version_id size in bytes.
+  private static final byte VERSION_ID_SIZE = 1;
+  private static final int BINARY_VERSION_TRACE_ID_OFFSET = VERSION_ID_SIZE;
+  private static final int BINARY_VERSION_SPAN_ID_OFFSET = VERSION_ID_SIZE + TraceId.SIZE;
+  private static final int BINARY_VERSION_TRACE_OPTIONS_OFFSET =
+      VERSION_ID_SIZE + TraceId.SIZE + SpanId.SIZE;
+  private static final int BINARY_VERSION_FORMAT_LENGTH =
+      VERSION_ID_SIZE + TraceId.SIZE + SpanId.SIZE + TraceOptions.SIZE;
+  private static final int HEX_VERSION_FORMAT_LENGTH = 2 * BINARY_VERSION_FORMAT_LENGTH;
+
+  // Disallow instances of this class.
+  private HttpPropagationUtil() {}
+
+  /**
+   * Returns the header name that must be used in the HTTP request for the tracing context.
+   *
+   * @return the header name that must be used in the HTTP request for the tracing context.
+   */
+  public static String getHttpHeaderName() {
```

Comment:
Since this is just a constant string, why not make the value itself public, and skip the method?